barretenberg/cpp/src/barretenberg/vm/avm_trace/avm_trace.cpp

```diff
@@ -1365,13 +1393,23 @@ void AvmTraceBuilder::op_nullifier_exists(uint32_t note_offset, uint32_t dest_of
     main_trace.push_back(row);
 }
 
-void AvmTraceBuilder::op_sload(uint32_t slot_offset, uint32_t value_offset)
+void AvmTraceBuilder::op_sload(uint32_t slot_offset, uint32_t write_offset)
 {
     auto const clk = static_cast<uint32_t>(main_trace.size());
 
-    Row row =
-        create_kernel_output_opcode_with_metadata(clk, value_offset, AvmMemoryTag::FF, slot_offset, AvmMemoryTag::FF);
-    kernel_trace_builder.op_sload(clk, row.avm_main_ib, row.avm_main_ia);
+    // Read the slot
+    AvmMemTraceBuilder::MemRead slot_read = mem_trace_builder.read_and_load_from_memory(
+        call_ptr, clk, IntermRegister::IB, slot_offset, AvmMemoryTag::FF, AvmMemoryTag::FF);
+
+    // Get the data value from the execution_hints
+    // TODO: for now the hints are being offset by the offset - this will NOT fly, but i struggled to get the hash
+    // working for FF
+    FF value = execution_hints.storage_values.at(write_offset);
+    // TODO: throw error if the hint does not exist
+
+    Row row = create_sload(clk, write_offset, value, slot_read.val, slot_offset);
+    kernel_trace_builder.op_sload(clk, row.avm_main_ib, value);
+
     row.avm_main_sel_op_sload = FF(1);
 
     main_trace.push_back(row);
```

Comment:
My attempt at understanding this:

1. Use the instruction's `slot_offset` operand and read `slot` from memory (adds row to memory trace)
2. Use that `slot` to retrieve a storage `value` from `execution_hints.storage_values`
3. Write the hinted `value` into memory (adds row to memory trace)
4. Add row to kernel trace for this storage slot->value "kernel output lookup"
5. Create a new row in the main trace for the sload, including mem read for slot, mem write for value, and flag to indicate presence of a "kernel output lookup"
6. Activate the selector for sload in this row

Reply:
Yep this is pretty spot on